Feature: OpenAPI 3.0 docs

machisuji opened the pull request:
**IMPORTANT: Do _not_ squash.** _The history here is important as it contains the code for converting from API Blueprint to OpenAPI 3.0. We probably don't need it anymore. But if we do it would be unfortunate not to have it anymore._

Replaces API Blueprint docs (.apib files) with OpenAPI 3.0 ones (.yml), exposes spec under /api/v3/spec.json and adds swagger ui under /api/docs to try out the API.

We use the `openapi-cli` linter to validate the spec in a github workflow which should make sure no invalid spec is committed.
There are still 14 (mostly unknown types) errors (and a lot of warnings) present at this point, however. Fixing those is out-of-scope for this PR. Hence they've been added to `.redocly.lint-ignore.yaml` for now.

Either way the spec is good enough to be used by the integrated SwaggerUI. We still want to fix all errors and warnings eventually.

machisuji committed:
OpenAPI 3.0 spec, swagger UI

right now the spec is generated from the existing api blueprint docs

machisuji committed:
openapi pre, pre apibp conversion

machisuji committed:
openapi spec, post apib conversion

machisuji committed:
/api/docs finished

- made into separate, lazy loaded js/css bundle in angular
- added setting allowing to disable API docs
- added spec lint workflow
- fixed styling (line-height)
- renamed/restructured

b12f commented on app/views/api_docs/index.html.erb:
```diff
@@ -0,0 +1 @@
+FOOBAR
```
This file can be deleted I think

b12f commented on frontend/src/app/features/api-docs/swagger-ui/swagger-ui.component.html:
```diff
@@ -0,0 +1,3 @@
+<div id="docs-wrapper">
```
Do you need the wrapper? I think you might even be able to have no template at all, and instead do something like

```
@Component({
  // other stuff
  // ...
  host: {
    id: 'swagger',
  },
})
```

Which would reduce both the complexity of the code but most importantly of the resulting html

machisuji replied:
Tried this but it doesn't seem to work. The compiler complains about a missing template. Leaving it as it is in that case.

b12f reviewed (commented):
lgtm, a couple of small notes

machisuji committed:
addressed review (including code climate) remarks

machisuji committed:
put spec assembly into independent script

as in it doesn't require bundling or a database
conncetion as opposed to a rake task

machisuji committed:
put spec validation in script; polish

machisuji committed:
amended executable flag

machisuji committed:
added missing ruby dep to workflow

machisuji committed:
ruby/setup-ruby doesn't work on self-hosted runner

error: The environment variable ImageOS must be set

machisuji committed:
use same ruby version as code

machisuji committed:
removed superfluous %x

machisuji committed:
code climate

machisuji committed:
merged view_user_model into user_model
